content/en/blog/2021/blog-mermaid.md

```diff
@@ -0,0 +1,170 @@
+
+The Mermaid documentation is excellent - simple text, multiple examples and diagrams included!
+
+---
+
+## What do I need to start using Mermaid?
+
+You need your favorite markdown editor, live editor access and familiarity with Hugo. 
+
+## Examples
+
+The first diagram is shown in [What is Ingress?](https://kubernetes.io/docs/concepts/services-networking/ingress/#what-is-ingress)
+
+![ingress-svg](/img/ingress-figure.svg)
+
+<p style="text-align: center; font-weight: bold"><a href="https://mermaid-js.github.io/mermaid-live-editor/edit/#eyJjb2RlIjoiZ3JhcGggIExSXG4gIGNsaWVudChbY2xpZW50XSktLiBJbmdyZXNzLW1hbmFnZWQgPGJyPiBsb2FkIGJhbGFuY2VyIC4tPmluZ3Jlc3NbSW5ncmVzc107XG4gIGluZ3Jlc3MtLT58cm91dGluZyBydWxlfHNlcnZpY2VbU2VydmljZV07XG4gIHN1YmdyYXBoIGNsdXN0ZXJcbiAgaW5ncmVzcztcbiAgc2VydmljZS0tPnBvZDFbUG9kXTtcbiAgc2VydmljZS0tPnBvZDJbUG9kXTtcbiAgZW5kXG4gIGNsYXNzRGVmIHBsYWluIGZpbGw6I2RkZCxzdHJva2U6I2ZmZixzdHJva2Utd2lkdGg6NHB4LGNvbG9yOiMwMDA7XG4gIGNsYXNzRGVmIGs4cyBmaWxsOiMzMjZjZTUsc3Ryb2tlOiNmZmYsc3Ryb2tlLXdpZHRoOjRweCxjb2xvcjojZmZmO1xuICBjbGFzc0RlZiBjbHVzdGVyIGZpbGw6I2ZmZixzdHJva2U6I2JiYixzdHJva2Utd2lkdGg6MnB4LGNvbG9yOiMzMjZjZTU7XG4gIGNsYXNzIGluZ3Jlc3Msc2VydmljZSxwb2QxLHBvZDIgazhzO1xuICBjbGFzcyBjbGllbnQgcGxhaW47XG4gIGNsYXNzIGNsdXN0ZXIgY2x1c3RlcjtcbiIsIm1lcm1haWQiOiJ7XG4gIFwidGhlbWVcIjogXCJkZWZhdWx0XCJcbn0iLCJ1cGRhdGVFZGl0b3IiOmZhbHNlLCJhdXRvU3luYyI6dHJ1ZSwidXBkYXRlRGlhZ3JhbSI6ZmFsc2V9">Live editor link: Ingress</a></p>
+
+Code:
+
+```mermaid
+graph LR;
+ client([client])-. Ingress-managed <br> load balancer .->ingress[Ingress];
+ ingress-->|routing rule|service[Service];
+ subgraph cluster
+ ingress;
+ service-->pod1[Pod];
+ service-->pod2[Pod];
+ end
+ classDef plain fill:#ddd,stroke:#fff,stroke-width:4px,color:#000;
+ classDef k8s fill:#326ce5,stroke:#fff,stroke-width:4px,color:#fff;
+ classDef cluster fill:#fff,stroke:#bbb,stroke-width:2px,color:#326ce5;
+ class ingress,service,pod1,pod2 k8s;
+ class client plain;
+ class cluster cluster;
+```
+
+---
+<br>
+
+Did you notice that some of the diagram elements are colored blue? That is accomplished using a `classDef` statement to define a class of style attributes, and a `class` statement to define the elements to apply that class to.
+
+In the code for the diagram above, we have examples of both: 
+```
+classDef k8s fill:#326ce5,stroke:#fff,stroke-width:4px,color:#fff; // defines style for the k8s class
+class ingress,service,pod1,pod2 k8s; // k8s class is applied to elements ingress, service, pod1 and pod2.
+```
+
+You can even style an individual element in your diagram using the style statement.
+```mermaid
+style box1 fill:#FF0000,stroke:333,stroke-width:3px // applies style to box1
+```
+
+---
+<br>
+
+The next diagram appears in [Pod Topology Spread Constraints](https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/#example-one-topologyspreadconstraint).
+
+![pod-spread](/img/pod-topo-constraint-02.svg)
```

Comment:
Here's an example of using a figure shortcode
```suggestion
{{< figure src="/img/pod-topo-constraint-02.svg" title="Example: One TopologySpreadConstraint " caption="Example: One TopologySpreadConstraint (rendered by Mermaid)" link="https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/#example-one-topologyspreadconstraint" >}}
```

Reply:
Got it, will replace with shortcode.